### PR TITLE
Add support for NordicTrack Z1300i

### PR DESCRIPTION
**Pull request by JosePadron**
Related to: https://github.com/cagnulein/qdomyos-zwift/issues/1805

- [x] Add support for NordicTrack Z1300i
- [ ] Bench test

### CODE REVIEW

File: src/proformtreadmill.cpp [hunk 1]
```diff
@@ -100,7 +102,7 @@ void proformtreadmill::forceIncline(double incline) {
         for (uint8_t i = 0; i < 7; i++) {
             write[14] += write[i + 6];
         }
-    } else {
+    } else { // nordictrack_z1300i_treadmill
```
Comment (JosePadron) [inline src/proformtreadmill.cpp:105]: Byte 14 is increased by 0x12

File: src/proformtreadmill.cpp [hunk 2]
```diff
@@ -148,7 +152,7 @@ void proformtreadmill::forceSpeed(double speed) {
...
```
Comment (JosePadron) [inline src/proformtreadmill.cpp:155]: Byte 14 is increased by 0x12

**Comment by cagnulein** [comment]
CI approved, let's wait the build

File: src/proformtreadmill.cpp [hunk 3]
```diff
@@ -819,7 +825,7 @@ void proformtreadmill::update() {
             if (counterPoll > 5) {
                 counterPoll = 0;
             }
-        } else if (nordictrack_s30_treadmill) {
+        } else if (nordictrack_s30_treadmill || nordictrack_z1300i_treadmill) {
```
Comment (JosePadron) [inline src/proformtreadmill.cpp:828]: Not sure about what data is needed. Model S30 keeps connection alive and displays data from trainer. Data could be the same. Could you check bluetooth logs to verify this?

Comment (cagnulein) [reply src/proformtreadmill.cpp:828]: Ok let's try first the build, then I will check the log tomorrow. Let me know

File: src/proformtreadmill.cpp [hunk 4]
```diff
@@ -1226,7 +1234,7 @@ void proformtreadmill::characteristicChanged(const QLowEnergyCharacteristic &cha
          (newValue.at(4) != 0x02 || (newValue.at(5) != 0x31 && newValue.at(5) != 0x34))) ||
         ((norditrack_s25i_treadmill) && (newValue.at(4) != 0x02 || (newValue.at(5) != 0x2f))) ||
         ((nordictrack_t65s_treadmill || proform_pro_1000_treadmill || nordictrack_t65s_83_treadmill || nordictrack_s30_treadmill ||
-          proform_treadmill_se || proform_cadence_lt) &&
+          nordictrack_z1300i_treadmill || proform_treadmill_se || proform_cadence_lt) &&
```
Comment (JosePadron) [inline src/proformtreadmill.cpp:1237]: Not sure about this one; but all nordic treadmills are here.

File: src/proformtreadmill.cpp [hunk 5]
```diff
@@ -2416,7 +2426,7 @@ void proformtreadmill::btinit() {
         QThread::msleep(sleepms);
         writeCharacteristic(noOpData3, sizeof(noOpData3), QStringLiteral("init"), false, false);
         QThread::msleep(sleepms);
-    } else if (nordictrack_s30_treadmill) {
+    } else if (nordictrack_s30_treadmill || nordictrack_z1300i_treadmill) {
```
Comment (JosePadron) [inline src/proformtreadmill.cpp:2429]: Same?

**Comment by cagnulein** [comment]
argh i thought I approved it yesterday for the CI. Done now, sorry. Let me know when you will try this

**Comment by cagnulein** [comment]
@JosePadron did you try the build?

**Comment by JosePadron** [comment]
> @JosePadron did you try the build?

Windows and IOs build are failing. Android job was fine. Where is the artifact? I cannot find link to download the .apk

**Comment by cagnulein** [comment]
here https://github.com/cagnulein/qdomyos-zwift/suites/18423286881/artifacts/1068755652

**Comment by JosePadron** [comment]
I just discovered that `Proform z1300i` works with the `NordicTrack z1300i`. No software change is required; but it would be nice if the label in the app gets updated. Request: https://github.com/cagnulein/qdomyos-zwift/pull/1817

Closing this request.

**Comment by cagnulein** [comment]
Glad to hear it!

Il giorno ven 24 nov 2023 alle 23:24 Jose Padron ***@***.***>
ha scritto:

**Pull request closed without merging**